tests/unit/type/list-3.tcl

```diff
@@ -47,39 +100,42 @@ foreach comp {2 1 0} {
     
     test "Stress tester for #3343-alike bugs comp: $comp" {
         r del key
+        set sent {}
+        set print_commands false
         for {set j 0} {$j < $cycles} {incr j} {
-            set op [randomInt 7]
-            set small_signed_count [expr 5-[randomInt 10]]
-            if {[randomInt 2] == 0} {
-                set ele [randomInt 1000]
-            } else {
-                set ele [string repeat x [randomInt 10000]][randomInt 1000]
-            }
-            switch $op {
-                0 {r lpush key $ele}
-                1 {r rpush key $ele}
-                2 {r lpop key}
-                3 {r rpop key}
-                4 {
-                    catch {r lset key $small_signed_count $ele}
-                }
-                5 {
-                    set otherele [randomInt 1000]
-                    if {[randomInt 2] == 0} {
-                        set where before
-                    } else {
-                        set where after
-                    }
-                    r linsert key $where $otherele $ele
-                }
-                6 {
-                    set index [randomInt [r llen key]]
-                    set otherele [r lindex key $index]
-                    r lrem key 1 $otherele
+            if {[catch {
+                set cmd [generate_cmd_on_list_key key]
+                lappend sent $cmd
+
+                # execute the command, we expect commands to fail on syntax errors
+                r {*}$cmd
+            }]} {
+                if {[count_log_message 0 "crashed by signal"] != 0 || [count_log_message 0 "ASSERTION FAILED"] != 0} {
```

Comment:
the search for `ASSERTION FAILED` is insufficient, a serverPanic prints something else.
how about a PING check instead instead?

Reply:
Originally I was using ping, but it affected the speed of the test, I changed to read log files, it seems that ping will be more reasonable.

Reply:
there are many other ways to check if the process is alive (like `kill -0`) but it'll still not support external mode.
I think we don't have to add ping after every command (or even after every command that fails).
i think it is enough to do when the test ends.
i.e. we run a sequence of commands, and for all of them we use `catch` so we have no idea if the command fails because of execution issues or because of a crash.
but either way, when the test ends, if the server is not alive (ping test) we print the full list of commands.
that's what i do in the corrupt-dump-fuzzer